.github/workflows/python-code-checks.yaml

```diff
@@ -0,0 +1,24 @@
+name: Run code checks
+
+on:
+    # Trigger it when a new pull request is opened.
+    pull_request:
+
+    # Trigger it  when pushing to the master branch.
+    push:
+        branches:
+            - master
+        tags-ignore:
+            - "**" # Ignore all tags to prevent duplicate checks when tags are pushed.
+
+    # Allow manual trigger.
+    workflow_dispatch:
+
+jobs:
+    lint_check:
+        name: Lint check
+        uses: apify/workflows/.github/workflows/python_lint_check.yaml@main
+
+    type_check:
+        name: Type check
+        uses: apify/workflows/.github/workflows/python_type_check.yaml@main
```

Comment:
I have to say, I'm not the biggest fan of adding this `Makefile` just for linting the python bindings :sweat_smile:  but I suppose there isn't really much we can do right now, right? 

What if added some inputs to the `python_lint_check` (`working_directory`...), and made it call `uv sync`, `ruff`...  directly (instead of `make install-dev`...)? Have you guys already thought of this (are there any roadblocks)?